client/faucet/src/mint.rs

```diff
@@ -27,13 +32,34 @@ impl std::fmt::Display for Response {
     }
 }
 
+const HEX_CHARSET: &[u8] = b"abcdef0123456789";
+
+fn random_hex_string(len: usize) -> String {
+    let mut rng = thread_rng();
+    (0..len)
+        .map(|_| HEX_CHARSET[rng.gen_range(0..HEX_CHARSET.len())] as char)
+        .collect()
+}
+
+fn random_uuid() -> String {
+    format!(
+        "{}-{}-{}-{}-{}",
+        random_hex_string(8),
+        random_hex_string(4),
+        random_hex_string(4),
+        random_hex_string(4),
+        random_hex_string(12)
+    )
+}
+
```

Comment:
I'd recommend using the `uuid` crate for this instead of reinventing this logic here.